src/Stan.hs

```diff
@@ -50,17 +57,21 @@ runStan StanArgs{..} = do
                 infoMessage " 💡 Try using --cabal-file-path option to specify the path to the .cabal file.\n"
                 pure mempty
 
-    let analysis = runAnalysis extensionsMap hieFiles
+    let analysis = runAnalysis cabalExtensionsMap hieFiles
     putTextLn $ prettyShowAnalysis analysis stanArgsToggleSolution
 --    debugHieFile "target/Target/Infinite.hs" hieFiles
   where
     getExtensionsWithCabal :: FilePath -> [HieFile] -> IO (Map FilePath ExtensionsResult)
     getExtensionsWithCabal cabal hies = do
-        -- create Map from filepathes to sources.
-        let sourcesMap = fromList $ map (\HieFile{..} -> (hie_hs_file, hie_hs_src)) hies
-
         infoMessage $ "Using the following .cabal file: " <> toText cabal <> "\n"
-        getPackageExtentionsBySources cabal sourcesMap
+        (Right . Set.fromList <<$>> parseCabalFileExtensions cabal)
+            `catch` handleCabalErr
+      where
+        handleCabalErr :: CabalException -> IO (Map FilePath ExtensionsResult)
+        handleCabalErr _ = do
+            errorMessage "Error when parsing cabal file. Stan will continue without information from .cabal file"
+            pure $ Map.fromList $
+                map (mapToSnd (const $ Left CabalParseError) . hie_hs_file) hies
```

Comment:
As we discussed, it should be possible to add a constructor to the `ExtensionError` type to store `CabalParseError`, so we can fill the map with the specific `CabalException` we get in this function. Do you think it should be done in the `extensions` library?

Reply:
Yes, we can improve the experience in there as well :+1: